packages/core/util/jexl.ts

```diff
@@ -6,133 +6,111 @@ type JexlWithAddFunction = typeof jexl & {
-    (str: string, targetLength: number, fillString?: string | undefined) => {
-      str.padStart(targetLength, fillString)
-    },
+    (str: string, targetLength: number, fillString?: string | undefined) =>
+      str.padStart(targetLength, fillString),
   )
-  jexlInstance.addTransform('repeat', (str: string, count: number) => {
-    str.repeat(count)
-  })
-  jexlInstance.addTransform(
-    'replace',
-    (str: string, match: string, newSubStr: string) => {
-      str.replace(match, newSubStr)
-    },
+  j.addFunction('repeat', (str: string, count: number) => str.repeat(count))
+  j.addFunction('replace', (str: string, match: string, newSubStr: string) =>
+    str.replace(match, newSubStr),
   )
-  jexlInstance.addTransform(
-    'replaceAll',
-    (str: string, match: string, newSubStr: string) => {
-      str.replaceAll(match, newSubStr)
-    },
+  j.addFunction('replaceAll', (str: string, match: string, newSubStr: string) =>
+    str.replaceAll(match, newSubStr),
   )
-  jexlInstance.addTransform(
+  j.addFunction(
     'slice',
-    (str: string, start: number, end?: number | undefined) => {
-      str.slice(start, end)
-    },
+    (str: string, start: number, end?: number | undefined) =>
+      str.slice(start, end),
   )
-  jexlInstance.addTransform(
+  j.addFunction(
     'startsWith',
-    (str: string, searchStr: string, position?: number | undefined) => {
-      str.startsWith(searchStr, position)
-    },
+    (str: string, searchStr: string, position?: number | undefined) =>
+      str.startsWith(searchStr, position),
   )
-  jexlInstance.addTransform(
+  j.addFunction(
     'substring',
-    (str: string, start: number, end?: number | undefined) => {
-      str.substring(start, end)
-    },
+    (str: string, start: number, end?: number | undefined) =>
+      str.substring(start, end),
   )
-  jexlInstance.addTransform('toLowerCase', (str: string) => {
-    str.toLowerCase()
-  })
-  jexlInstance.addTransform('toUpperCase', (str: string) => {
-    str.toUpperCase()
-  })
-  jexlInstance.addTransform('trim', (str: string) => {
+  j.addFunction('toLowerCase', (str: string) => str.toLowerCase())
+  j.addFunction('toUpperCase', (str: string) => str.toUpperCase())
+  j.addFunction('trim', (str: string) => {
     str.trim()
   })
-  jexlInstance.addTransform('trimEnd', (str: string) => {
-    str.trimEnd()
-  })
-  jexlInstance.addTransform('trimStart', (str: string) => {
-    str.trimStart()
-  })
+  j.addFunction('trimEnd', (str: string) => str.trimEnd())
+  j.addFunction('trimStart', (str: string) => str.trimStart())
 
-  jexlInstance.addTransform('hashcode', (str: string) => {
-    return str
+  j.addFunction('hashcode', (str: string) =>
+    str
       .split('')
       .map(c => c.charCodeAt(0))
-      .reduce((a, b) => a + b, 0)
+      .reduce((a, b) => a + b, 0),
+  )
+
+  j.addFunction('getTag', (feature: Feature, str: string) => {
+    const tags = feature.get('tags')
+    return tags ? tags[str] : feature.get(str)
```

Comment:
this could be written in jexl format without having it's own add function in jexl.ts
something like jexl:feature.get('tags') ? tags[str] : feature.get(str) assuming str is in the context variables

Reply:
could be written in jexl, yes, but this is a nice convenience